python/lsst/daf/butler/registry/_remoteRegistry.py

```diff
@@ -0,0 +1,431 @@
+
+    def findDataset(self, datasetType: Union[DatasetType, str], dataId: Optional[DataId] = None, *,
+                    collections: Any = None, timespan: Optional[Timespan] = None,
+                    **kwargs: Any) -> Optional[DatasetRef]:
+        # Docstring inherited from lsst.daf.butler.registry.Registry
+        raise NotImplementedError()
+
+    def insertDatasets(self, datasetType: Union[DatasetType, str], dataIds: Iterable[DataId],
+                       run: Optional[str] = None, expand: bool = True,
+                       idGenerationMode: DatasetIdGenEnum = DatasetIdGenEnum.UNIQUE) -> List[DatasetRef]:
+        # Docstring inherited from lsst.daf.butler.registry.Registry
+        raise NotImplementedError()
+
+    def _importDatasets(self, datasets: Iterable[DatasetRef], expand: bool = True,
+                        idGenerationMode: DatasetIdGenEnum = DatasetIdGenEnum.UNIQUE,
+                        reuseIds: bool = False) -> List[DatasetRef]:
+        # Docstring inherited from lsst.daf.butler.registry.Registry
+        raise NotImplementedError()
+
+    def getDataset(self, id: DatasetId) -> Optional[DatasetRef]:
+        # Docstring inherited from lsst.daf.butler.registry.Registry
+        raise NotImplementedError()
+
+    def removeDatasets(self, refs: Iterable[DatasetRef]) -> None:
+        # Docstring inherited from lsst.daf.butler.registry.Registry
+        raise NotImplementedError()
+
+    def associate(self, collection: str, refs: Iterable[DatasetRef]) -> None:
+        # Docstring inherited from lsst.daf.butler.registry.Registry
+        raise NotImplementedError()
+
+    def disassociate(self, collection: str, refs: Iterable[DatasetRef]) -> None:
+        # Docstring inherited from lsst.daf.butler.registry.Registry
+        raise NotImplementedError()
+
+    def certify(self, collection: str, refs: Iterable[DatasetRef], timespan: Timespan) -> None:
+        # Docstring inherited from lsst.daf.butler.registry.Registry
+        raise NotImplementedError()
+
+    def decertify(self, collection: str, datasetType: Union[str, DatasetType], timespan: Timespan, *,
+                  dataIds: Optional[Iterable[DataId]] = None) -> None:
+        # Docstring inherited from lsst.daf.butler.registry.Registry
+        raise NotImplementedError()
+
+    def getDatastoreBridgeManager(self) -> DatastoreRegistryBridgeManager:
+        """Return an object that allows a new `Datastore` instance to
+        communicate with this `Registry`.
+
+        Returns
+        -------
+        manager : `DatastoreRegistryBridgeManager`
+            Object that mediates communication between this `Registry` and its
+            associated datastores.
+        """
+        from ..tests._dummyRegistry import DummyDatastoreRegistryBridgeManager, DummyOpaqueTableStorageManager
+        return DummyDatastoreRegistryBridgeManager(DummyOpaqueTableStorageManager(),
+                                                   self.dimensions, int)
```

Comment:
I suspect I'll know more when I've looked at more of the PR, but I'm guessing it might make more sense to just move this method to `SqlRegistry` and not define it for `RemoteRegistry`.

Reply:
Yes, I added it here for now because without it *everything* blows up because Butler tries to create a Datastore class by passing in the manager and this requires a "real" datastore manager even if Datastore is not being used. Without this I couldn't even demonstrate the `butler query-X` command lines worked.